src/utils.rs

```diff
@@ -56,6 +62,26 @@ impl NibbleKey {
     pub fn len(&self) -> usize {
         self.0.len()
     }
+
+    pub fn with_hex_prefix(&self, is_terminator: bool) -> Vec<u8> {
```

Comment:
// This method is adding the hex prefix AND converting the nibble key to a byte key.